src/components/Todos/Todos.tsx

```diff
@@ -0,0 +1,136 @@
+import cn from 'classnames';
+import React, { useEffect, useRef, useState } from 'react';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todos: Todo[],
+  deleteTodo: (id: number) => void,
+  updatingTodos: number[],
+  updateTodo: (todoId: number, todoData: Partial<Todo>) => void,
+};
+
+export const Todos: React.FC<Props> = ({
+  todos,
+  deleteTodo,
+  updatingTodos,
+  updateTodo,
+}) => {
+  const [editTodoId, setEditTodoId] = useState<number | null>(null);
+  const [titleValue, setTitleValue] = useState<string>('');
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, [editTodoId]);
+
+  const handleTitleUpdate = (
+    id: number,
+    todoTitle: string,
+  ) => {
+    if (!todoTitle.length) {
+      deleteTodo(id);
+      setEditTodoId(null);
+
+      return;
+    }
+
+    if (todoTitle !== titleValue) {
+      updateTodo(id, { title: titleValue });
+    }
+
+    setEditTodoId(null);
+  };
+
+  return (
+    <section className="todoapp__main">
+      {todos.map(({
+        id,
+        title,
+        completed,
+      }) => {
+        return (
+          <div
+            className={cn('todo', { completed })}
+            key={id}
+          >
+            <label className="todo__status-label">
+              <input
+                type="checkbox"
+                className="todo__status"
+                defaultChecked={completed}
+                onChange={() => updateTodo(
+                  id,
+                  { completed: !completed },
+                )}
+              />
+            </label>
+
+            {id === editTodoId ? (
+              <>
+                <form
+                  onSubmit={(e) => {
+                    e.preventDefault();
+                    handleTitleUpdate(id, title);
+                  }}
+                >
```

Comment:
i think this is the root cause
```suggestion
                <form
                  onSubmit={(e) => {
                    e.preventDefault();
                    e.target.value
                      ? handleTitleUpdate(id, title)
                      : deleteTodo(id);
                  }}
                >
```

Reply:
✅ fixed above!